src/mod_cache_users.erl

```diff
@@ -94,6 +91,7 @@ does_cached_user_exist(Status, _, _, _) ->
                                 RequestType :: ejabberd_auth:exist_type()) ->
     boolean().
 maybe_put_user_into_cache(true, HostType, Jid, stored) ->
-    mongoose_user_cache:merge_entry(HostType, ?MODULE, Jid, #{});
+    mongoose_user_cache:merge_entry(HostType, ?MODULE, Jid, #{}),
+    true;
```

Comment:
Yes, it could lead to returning false if the merge failed, which would then make the whole handler return the wrong value. Chances are extremely low, but well, not technically correct.

Reasons to fail a merge is some brutal contention. There's retries and everything, so I wouldn't expect that to ever happen, but, let's be correct 😄